File: resources/code/parse_and_print/parse_and_print.py
from libcellml import Model, Printer

# Create a Printer instance.
printer = Printer()

# Create a model to print.
model = Model("my_model")

# The output of the printModel function is a string representing the serialised input model.
serialised_model = printer.printModel(model)

# Check the printer for issues.
assert(printer.issueCount() == 0)

# Write the string to a file.
with open("my_printed_file.cellml", "w") as write_file:
    write_file.write(serialised_model)

File: resources/code/parse_and_print/parse_and_print.cpp
#include <cassert>
#include <fstream>
#include <libcellml>

int main(int argc, char *argv[])
{
  // Create a Printer instance.
  auto printer = libcellml::Printer::create();

  // Create a Model instance.
  auto model = libcellml::Model::create("my_model");

  // The output of the printModel function is a string representing the serialised input model.
  const std::string serialisedModelString = printer->printModel(model);

  // Check the printer for issues.
  assert(printer->issueCount() == 0);

  // Write the serialised string to a file.
  const std::string outFileName = "my_printed_file.cellml";
  std::ofstream outFile(outFileName);
  outFile << serialisedModelString;
  outFile.close();

  return 0;
}
